### PR TITLE
Added support for Btrfs quotas in YaST (jsc#SLE-15285)

### PR DESCRIPTION
**Pull request by tbazant**
### Description

YaST interface can now set size limits to Btrfs subvolumes. This update adds the procedure as well as screenshots.
Rendered PDF is attached (section 1.2.5)
[sec-filesystems-major-btrfs_color_en.pdf](https://github.com/SUSE/doc-sle/files/5666833/sec-filesystems-major-btrfs_color_en.pdf)

### Are there any relevant issues/feature requests?

* jsc#SLE-15285

### To which product versions do the changes apply?

The first column is to be filled by the requester, the second column is to be filled by the doc team after the changes have been backported to the maintenance branches.

| Code 15     | Backport Done
| ----------  | ---------- 
| [x] 15 SP3  | (`master` only)   
| [ ] 15 SP2  | [ ] 
| [ ] 15 SP1  | [ ] 
| [ ] 15 SP0  | [ ] 

| Code 12     | Backport Done
| ----------  | ---------- 
| [ ] 12 SP5  | [ ] 
| [ ] 12 SP4  | [ ] 
| [ ] 12 SP3  | [ ] 
| [ ] 12 SP2  | [ ]

### CODE REVIEW

File: xml/storage_filesystems.xml [hunk 1]
```diff
@@ -474,74 +474,171 @@ Id Path      single   single    single   Unallocated
...
+     </step>
+     <step>
+      <para>
+       In the main window, select the device for which you want to enable
+       subvolume quotas and click <guimenu>Edit</guimenu> at the bottom.
+      </para>
+     </step>
+     <step>
+      <para>
+       In the <guimenu>Edit Btrfs</guimenu> window, activate <guimenu>Enable
```
Comment (cjschroder) [inline xml/storage_filesystems.xml:512]: Not sure about "activate", "click" seems clearer.

File: xml/storage_filesystems.xml [hunk 2]
```diff
@@ -474,74 +474,171 @@ Id Path      single   single    single   Unallocated
...
+        </imageobject>
+        <imageobject role="html">
+         <imagedata fileref="yast2_btrfs_quotas_enable.png" width="75%"/>
+        </imageobject>
+       </mediaobject>
+      </figure>
+     </step>
+     <step>
+      <para>
+       From the list of existing subvolumes, activate the subvolume whose size
```
Comment (cjschroder) [inline xml/storage_filesystems.xml:529]: again, I suggest "click" instead of "activate"

File: xml/storage_filesystems.xml [hunk 3]
```diff
@@ -474,74 +474,171 @@ Id Path      single   single    single   Unallocated
...
+      <para>
+       Set a quota for one of the subvolumes that was listed in the previous
+       step. A subvolume can either be identified by path (for example
+       <filename>/var/tmp</filename>) or by <literal>0/<replaceable>SUBVOLUME
+       ID</replaceable></literal> (for example <literal>0/272</literal>). The
+       following example sets a quota of 5 GB for <filename>/var/tmp</filename>.
+      </para>
 <screen>&prompt.sudo;btrfs qgroup limit 5G /var/tmp</screen>
+      <para>
+       The size can either be specified in bytes (5000000000), kilobytes
```
Comment (cjschroder) [inline xml/storage_filesystems.xml:605]: Should the bytes value be 5000000000B?

**Review by cjschroder — CHANGES_REQUESTED** [review]
Nice work, I have a few comments.

File: xml/storage_filesystems.xml [hunk 4]
```diff
@@ -474,74 +474,171 @@ Id Path      single   single    single   Unallocated
...
-    quota support for subvolumes. If you set up the root file system by using
-    the respective &yast; proposal, it is prepared accordingly: quota groups
-    (<literal>qgroup</literal>) for all subvolumes are already set up. To set a
-    quota for a subvolume in the root file system, proceed as follows:
+    The Btrfs root file system subvolumes (for example,
+    <filename>/var/log</filename>, <filename>/var/crash</filename> or
+    <filename>/var/cache</filename>) can use all of the available disk space
+    during normal operation, and cause a system malfunction. To help avoid this
+    situation, &productname; offers quota support for Btrfs subvolumes. If you
+    set up the root file system by using the respective &yast; proposal, you
```
Comment (tbazant) [inline xml/storage_filesystems.xml:482]:
```suggestion
    set up the root file system from a &yast; proposal, you
```

[the remark this answers is not in the document]
Comment (tbazant) [reply xml/storage_filesystems.xml:482]: thanks for proposal :-)

File: xml/storage_filesystems.xml (hunk 1 above)
Comment (tbazant) [reply xml/storage_filesystems.xml:512]: When using check-boxes, somebody (Stefan) wired it in my head a long time ago that 'activate' is better than 'check', not sure though

File: xml/storage_filesystems.xml [hunk 5]
```diff
@@ -474,74 +474,171 @@ Id Path      single   single    single   Unallocated
...
+     <step>
+      <para>
+       In the main window, select the device for which you want to enable
+       subvolume quotas and click <guimenu>Edit</guimenu> at the bottom.
+      </para>
+     </step>
+     <step>
+      <para>
+       In the <guimenu>Edit Btrfs</guimenu> window, activate <guimenu>Enable
+       Subvolume Quotas</guimenu> and confirm with <guimenu>Next</guimenu>.
```
Comment (tbazant) [inline xml/storage_filesystems.xml:513]:
```suggestion
       In the <guimenu>Edit Btrfs</guimenu> window, activate the <guimenu>Enable
       Subvolume Quotas</guimenu> check-box and confirm with <guimenu>Next</guimenu>.
```

File: xml/storage_filesystems.xml (hunk 2 above)
Comment (tbazant) [inline xml/storage_filesystems.xml:529]:
```suggestion
       From the list of existing subvolumes, click the subvolume whose size
```

Comment (tbazant) [reply xml/storage_filesystems.xml:529]: here it's not a check-box, just highlighting a line with subvolume's name, click is better

File: xml/storage_filesystems.xml (hunk 3 above)
Comment (tbazant) [reply xml/storage_filesystems.xml:605]: as a basic measure, bytes can live without a unit specifier

**Review by tbazant — COMMENTED** [review]
included some suggestions

File: xml/storage_filesystems.xml (hunk 1 above)
Comment (cjschroder) [reply xml/storage_filesystems.xml]: ok!

File: xml/storage_filesystems.xml (hunk 5 above)
Comment (cjschroder) [reply xml/storage_filesystems.xml]: I like this.

File: xml/storage_filesystems.xml (hunk 3 above)
Comment (cjschroder) [reply xml/storage_filesystems.xml:605]: :+1:

File: xml/storage_filesystems.xml (hunk 4 above)
Comment (cjschroder) [reply xml/storage_filesystems.xml]: :+1:  Much better :)

File: xml/storage_filesystems.xml (hunk 2 above)
Comment (cjschroder) [reply xml/storage_filesystems.xml]: :+1:

**Comment by cjschroder** [comment]
Allrightythen, I'm not sure about the new buttons for "Commit suggestion" and "Add suggestion to batch", but I officially approve.

**Comment by ancorgs** [comment]
As a YaST developer involved in the YaST Partitioner implementation of quotas, this looks good to me. Thanks.

Just take into account we have received no feedback yet from the requester of the feature. So I cannot promise this will be the final implementation.

**Review by ancorgs — APPROVED** [review]
Approved. See my previous comment.

**Comment by cjschroder** [comment]
@tbazant I logged into Travis-CI and re-ran this build because it affects later commits, and it failed again "error: IDREF "sec-filesystems-major-btrfs-quota" without matching ID". I don't see why it is throwing this error, as `<sect2 xml:id="sec-filesystems-major-btrfs-quota">` exists. It builds without errors on my computer. I didn't have time to try to debug this, I am off for the holiday until Dec. 28. Sorry I couldn't do more.

**Comment by sknorr** [comment]
@cjschroder Ftr, the issue was in the SLED documentation (not the SLES documentation) and Toms later fixed that using profiling: https://github.com/SUSE/doc-sle/commit/313fa5b2ecd7082245ac563fcc3e09d711a452f0 .

**Pull request merged**